core/src/utils/FilesystemUtils.cpp

```diff
@@ -29,19 +29,31 @@
  *
  */
 
-#include "FilesystemUtils.h"
-#include <QDir>
-#include <QDebug>
-#include <QCoreApplication>
+#include "FilesystemUtils.hpp"
 #include <iostream>
+#include <experimental/filesystem>
 
-void ledger::qt::FilesystemUtils::clearFs(const std::string& path) {
-    QDir root(QString::fromStdString(path));
-    for (const auto& file : root.entryInfoList(QDir::AllEntries | QDir::NoDotAndDotDot)) {
-        if (file.isDir() && file.absoluteFilePath().compare(root.absolutePath()) != 0) {
-            clearFs(file.absoluteFilePath().toStdString());
-        } else if (!file.isExecutable()) {
-            QFile::remove(file.absoluteFilePath());
+using namespace std;
+namespace fs = std::experimental::filesystem::v1;
+
+bool ledger::core::FilesystemUtils::isExecutable(const std::string& path){
+    fs::path filePath{path};
+#ifdef _WIN32
+	auto extension=filePath.extension().string();
+    return (extension==".exe") || (extension==".bat") || (extension==".com")
+#else
+    return (status(filePath).permissions() & fs::perms::owner_exec)!=fs::perms::none;
```

Comment:
Only `owner_exec` ? Why not checking `group_exec` and `other_exec` as well ?

Reply:
fixed